app/src/main/java/com/github/se/polyfit/MainActivity.kt

```diff
@@ -8,24 +9,27 @@ import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.github.se.polyfit.ui.navigation.Navigation
 import com.github.se.polyfit.ui.navigation.Route
-import com.github.se.polyfit.ui.screen.HomeScreen
 import com.github.se.polyfit.ui.screen.LoginScreen
 import com.github.se.polyfit.ui.theme.PolyfitTheme
+import dagger.hilt.android.AndroidEntryPoint
+import dagger.hilt.android.HiltAndroidApp
 
+@AndroidEntryPoint
 class MainActivity : ComponentActivity() {
-
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
     setContent {
       PolyfitTheme {
         val navController = rememberNavController()
         val navigation = Navigation(navController)
-        NavHost(navController = navController, startDestination = Route.Register) {
-          composable(Route.Register) { LoginScreen(navigation) }
-
-          composable(Route.Home) { HomeScreen() }
+        NavHost(navController = navigation.getController(), startDestination = Route.Register) {
```

Comment:
its a bit unclear to me, why are we doing

```kt
val navController = rememberNavController()
val navigation = Navigation(navController)
```
only to get the controller back from 
```kt
navController = navigation.getController()
```

It seems redundant, is this something that needs to be done for the injection to work?